tests/test_edgeql_json.py

```diff
@@ -1378,6 +1381,43 @@ async def test_edgeql_json_slice_03(self):
                 SELECT to_json('[1, "a", 3, null]')[:'1'];
             """)
 
+    async def test_edgeql_json_slice_04(self):
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[:1];''',
+            ['h'],
+            ['"h"'],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[:-1];''',
+            ['hell'],
+            ['"hell"'],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[1:-1];''',
+            ['ell'],
+            ['"ell"'],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[-1:1];''',
+            [''],
+            ['""'],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[-100:100];''',
+            ['hello'],
+            ['"hello"'],
+        )
+
+        await self.assert_query_result(
+            r'''SELECT to_json('"hello"')[100:-100];''',
+            [''],
+            ['""'],
+        )
```

Comment:
I don't see error test cases, e.g.: 

`SELECT to_json('"hello"')[20]`

`SELECT to_json('["hello"]')[0][20]`

`SELECT to_json('["hello"]')[0][100]`

and also that empty slices like below returning empty strings

`SELECT to_json('"hello"')[:-100]`

`SELECT to_json('["hello"]')[0][100:]`